lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1213,6 +1216,23 @@ def _create_deployment_parameters(
                     if f.type not in features_settings:
                         features_settings[f.type] = f
 
+            # If Ultra Disk is being used, try to set availability zone to
+            # VM's capability.
+            arm_parameters.use_availability_sets = (
+                self._azure_runbook.use_availability_sets
+            )
+            if (
+                node.capability.disk.data_disk_type == schema.DiskType.UltraSSDLRS
+                and not arm_parameters.use_availability_sets
+                and not arm_parameters.availability_zones
+            ):
+                arm_parameters.use_availability_sets = False
+                arm_parameters.availability_zones = node.capability.zones.items
```

Comment:
@squirrelsc node.capability.zones is not being populated from the node_space created in _resource_sku_to_capability(). Do you know what I am doing wrong?

Reply:
Please remove the cached files, so the cache will be refreshed immediately.

Reply:
I did that as well. I see that the node_space in _resource_sku_to_capability() is set correctly. Still node.capability.zones is not set

Reply:
Ok, I got understand it now. The current logic updates the settings in create_deployment_parameters, it's not designed to save back to the disk. It needs to write in `_resource_sku_to_capability` as you mentioned, so it could be saved to the cached file, and be used next time.